runners/samza/src/main/java/org/apache/beam/runners/samza/util/SamzaPipelineExceptionListener.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.samza.util;
+
+/**
+ * An ExceptionListener following Observer pattern. Any runtime exception caught by {@code
+ * OpAdapter} will be notified to any concrete SamzaPipelineExceptionListener at Runtime
+ */
+public interface SamzaPipelineExceptionListener {
+
+  void onException(String opName, Exception exception);
```

Comment:
Instead of transformName, I suggest adding a new wrapper class, e.g. ExceptionContext, which includes transform name, That way we can pass more info down the road if needed, e.g. transform urn, payload, etc...

In design of an api, please think about future evolvement. One api for a use case is not that great.